lib/charms/kratos_external_idp_integrator/v0/kratos_external_provider.py

```diff
@@ -507,6 +509,30 @@ def remove_provider(self) -> None:
         for relation in self._charm.model.relations[self._relation_name]:
             relation.data[self._charm.app].clear()
 
+    def get_redirect_uri(self, relation_id: Optional[int] = None) -> str:
```

Comment:
This is only called without a relation ID right now so why support the `relation_id` parameter?

Reply:
The return type should be Optional[str], there are conditions here where there's simple `return`

Reply:
> This is only called without a relation ID right now so why support the `relation_id` parameter?

To separate the lib logic from the charm itself, I tried to make the library generic. In this case it is not necessary because it is highly unlikely that it will be used by other charms. In principle, I think that it makes sense to keep it that way. But if you wish I can change it.